File: frontend/src/services/staffService.js
import api from './api';

export const getAllStaff = () => api.get('/staff');
export const getStaffById = (id) => api.get(`/staff/${id}`);
export const createStaff = (staff) => api.post('/staff', staff);
export const updateStaff = (id, staff) => api.put(`/staff/${id}`, staff);
export const deleteStaff = (id) => api.delete(`/staff/${id}`);

File: backend/src/main/java/com/hospital/service/StaffService.java
package com.hospital.service;

import com.hospital.entity.Staff;
import java.util.List;

public interface StaffService {
    Staff createStaff(Staff staff);
    Staff getStaffById(Long id);
    List<Staff> getAllStaff();
    Staff updateStaff(Long id, Staff staffDetails);
    void deleteStaff(Long id);
}
